xskillscore/core/contingency.py

```diff
@@ -796,3 +796,223 @@ def _gerrity_s(table):
             dim=(OBSERVATIONS_NAME + "_category", FORECASTS_NAME + "_category"),
             skipna=True,
         ) / self._sum_categories("total")
+
+
+def roc(
+    observations,
+    forecasts,
+    bin_edges,
+    dim=None,
+    drop_intermediate=True,
+    return_results="area",
+):
+    """Computes the relative operating characteristic for a range of thresholds.
+
+    Parameters
+    ----------
+    observations : xarray.Dataset or xarray.DataArray
+        Labeled array(s) over which to apply the function.
+        If bin_edges=='continuous', observations are binary.
+    forecasts : xarray.Dataset or xarray.DataArray
+        Labeled array(s) over which to apply the function.
+        If bin_edges=='continuous', forecasts are probabilities.
+    bin_edges : array_like, str
+        Bin edges for categorising observations and forecasts.
+        Bins include the left most edge, but not the right.
+        If bin_edges='continuous', calculate bin_edges from forecasts, equal to
+        sklearn.metrics.roc_curve(f_boolean, o_prob)
+    dim : str, list
+        The dimension(s) over which to compute the contingency table
+    drop_intermediate : bool, default=True
+        Whether to drop some suboptimal thresholds which would not appear
+        on a plotted ROC curve. This is useful in order to create lighter
+        ROC curves.
+    return_results: str, default='area'
+        Specify how return is structed:
+            * 'area': return only the area under the curve of ROC
+            * 'all_as_tuple': return hit and false alarm rate at each bin and area under
+                the curve of ROC as tuple
+            * 'all_as_metric_dim': return hit and false alarm rate at each bin and area
+                under the curve of ROC concatinated into new `metric` dimension
+
+    Returns
+    -------
+    xarray.Dataset or xarray.DataArray : reduced by dimensions ``dim`` and specified by ``return_results``
```

Comment:
Can you be more specific here i.e. the variables and what they mean e.g. we do this in other variables e.g. we say it returns Pearson's correlation coefficient 
https://github.com/xarray-contrib/xskillscore/blob/master/xskillscore/core/deterministic.py#L104

and see https://github.com/scikit-learn/scikit-learn/blob/42aff4e2e/sklearn/metrics/_ranking.py#L862

Reply:
Sorry I was a bit confused originally
Could change to
```
xarray.Dataset or xarray.DataArray : reduced by dimensions ``dim``.
    See ``return_results`` parameter.
```